cmd/argocd/commands/headless/headless.go

```diff
@@ -109,6 +117,86 @@ func (c *forwardCacheClient) NotifyUpdated(key string) error {
 	})
 }
 
+func NewRepoClientset(namespace string, ctxStr string, clientOpts *apiclient.ClientOptions, kubeClientset *kubernetes.Clientset) repoapiclient.Clientset {
```

Comment:
Maybe split it somehow, because for embedded, you actually dont need all these params, and it may look a bit strange when you creating it

Reply:
agree, will do in another PR